src/Language/PureScript/Docs/Convert.hs

```diff
@@ -38,6 +102,95 @@ convertModule m@(P.Module _ coms moduleName  _ _) =
     >>> augmentDeclarations
     >>> map addDefaultFixity
 
+updateReExports ::
+  (Functor m, Applicative m,
+  MonadState (Map P.ModuleName (P.Module, Module)) m) =>
+  P.Env ->
+  [P.ModuleName] ->
+  m ()
+updateReExports env order = do
+  void $ traverse go order
+  where
+  go mn = do
+    (ast, mdl) <- lookup' mn
+    mdl' <- expandReExports env ast mdl
+    modify (Map.insert mn (ast, mdl'))
+
+  lookup' key = do
+    v <- gets (Map.lookup key)
+    case v of
+      Just v' ->
+        pure v'
+      Nothing ->
+        P.internalError ("Language.PureScript.Docs.Convert: missing key: " ++
+                        P.runModuleName key)
+
+-- We require that modules have already been sorted (P.sortModules) in order to
+-- ensure that by the time we convert a particular module, all its dependencies
+-- have already been converted.
+expandReExports ::
+  (Functor m, Applicative m,
+  MonadState (Map P.ModuleName (P.Module, Module)) m) =>
+  P.Env ->
+  P.Module ->
+  Module ->
+  m Module
+expandReExports env ast@(P.Module _ _ mn' _ mexports) mdl = do
+  let exports = fromMaybe (err "exports were not elaborated") mexports
+  let reExportNames = mapMaybe takeReExport exports
```

Comment:
One problem with this code is that it can't handle re-exported virtual modules (see also: `examples/docs/Virtual.purs`: https://github.com/hdgarrood/purescript/blob/reexport-docs/examples/docs/src/Virtual.purs, which, with this code, would cause an internal compiler error).

I want to get a list of all the (real) modules from which values are being re-exported, but I don't want the modules they were originally exported from; I want the modules they were imported from the current module's point of view, that is, I don't want it to follow the whole path of re-exports of re-exports.

This seems to work but of course it chokes on virtual modules. I'm not particularly familiar with this part of the compiler, so I'd appreciate some guidance.